operator/src/main/java/oracle/kubernetes/operator/helpers/ServiceHelper.java

```diff
@@ -528,9 +535,33 @@ private Step createNewService(Step next) {
 
     private Step deleteAndReplaceService(Step next) {
       V1DeleteOptions deleteOptions = new V1DeleteOptions();
+      if (getSpecType().equals(NODE_PORT_TYPE)) {
+        return deleteAndReplaceNodePortService();
+      } else {
+        return new CallBuilder()
+            .deleteServiceAsync(
+                createServiceName(), getNamespace(), getDomainUid(), deleteOptions, new DeleteServiceResponse(next));
+      }
+    }
+
+    private Step deleteAndReplaceNodePortService() {
       return new CallBuilder()
-          .deleteServiceAsync(
-              createServiceName(), getNamespace(), getDomainUid(), deleteOptions, new DeleteServiceResponse(next));
+              .withLabelSelectors(forDomainUidSelector(info.getDomainUid()), getCreatedbyOperatorSelector())
+              .listServiceAsync(
+                      getNamespace(),
+                      new ActionResponseStep<V1ServiceList>() {
+                      public Step createSuccessStep(V1ServiceList result, Step next) {
+                        Collection<V1Service> c = Optional.ofNullable(result).map(list -> list.getItems().stream()
```

Comment:
We are only supposed to have one of these service for a particular domain, so in theory we should check that and only delete and replace that one service.

Reply:
I was thinking if we should check if there is only one service that qualifies the checks in the end and only remove that one service just to be safe, so we don't accidentally remove and replace other services. But on the other hand, we don't know if we can do anything different if we do end up having multiple qualified services. I am fine with this then.